### PR TITLE
DEVX-5446 Implement Force Mute feature

### PR DESCRIPTION
**Pull request by superchilled**
## Description

Implements the force mute feature, either for an individual stream or for all streams in a session. Two methods have been added, one for each of these actions.

## Motivation and Context

It implements a feature in the SDK which has already been added to the API.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: lib/opentok/connections.rb [hunk 1]
```diff
@@ -24,5 +24,35 @@ def forceDisconnect(session_id, connection_id )
...
+    # This is an optional property. If you omit this property, all streams in the session will be muted.
+    # @example
+    # {
+    #   "active": true,
+    #   "excluded_streams": [
+    #     "excludedStreamId1",
+    #     "excludedStreamId2"
+    #   ]
+    # }
+
```
Comment (superchilled) [inline lib/opentok/connections.rb:52]: @jeffswartz I have a question about the `active` option. The [API Reference](https://tokbox.com/developer/rest/#force_mute_session) lists it as a mandatory field, but the [API Review](https://github.com/opentok/api-review/tree/master/mute-all-api#http-endpoint) lists it as a 'future field'.

During testing, I was able to make a request to the endpoint without this field being specified in the JSON payload.

Comment (jeffswartz) [reply lib/opentok/connections.rb:52]: @superchilled My understanding is that it is a required property.

**Review by jeffswartz — CHANGES_REQUESTED** [review]
@superchilled These methods do not relate to OpenTok connections. They relate to sessions and streams. (You mute streams in a session, not connections.) I propose that we should have the following two methods:

* `opentok.sessions.forceMuteAll(session_id, opts)`
* `opentok.streams.forceMute(session_id, stream_id,opts)`

I'm not a Ruby expert, so please let me know if this makes sense.

**Comment by superchilled** [comment]
> @superchilled These methods do not relate to OpenTok connections. They relate to sessions and streams. (You mute streams in a session, not connections.) I propose that we should have the following two methods:
> 
>     * `opentok.sessions.forceMuteAll(session_id, opts)`
> 
>     * `opentok.streams.forceMute(session_id, stream_id,opts)`
> 
> 
> I'm not a Ruby expert, so please let me know if this makes sense.

@jeffswartz ah, ok. I think I confused connections and streams. In that case, based on the current implementation, I think it would make more sense to define *both* methods in the `Streams` [class](https://github.com/opentok/OpenTok-Ruby-SDK/blob/main/lib/opentok/streams.rb). This would be in line with the fact that this class currently implements `Streams#find` (which hits the `https://api.opentok.com/v2/project/<apiKey>/session/<sessionId>/stream/<streamId>` endpoint) and `Streams#all` (which hits the `https://api.opentok.com/v2/project/<apiKey>/session/<sessionId>/stream/` endpoint). 

Implementing `forceMuteAll` in `Session` would be tricky because the `Session` object doesn't have direct access to the `Client` object (all calls to the API endpoints are called via methods in the `Client` object). The `Session` object basically just contains information about the current session and is used by other objects in the library when they need to invoke methods on the `Client`which relate to that session.

Let me know if you're happy for both these methods to be in the `Streams` class and I'll make the necessary changes.

**Review by jeffswartz — CHANGES_REQUESTED** [review]

File: lib/opentok/opentok.rb [hunk 2]
```diff
@@ -207,7 +207,7 @@ def signals
       @signals ||= Signals.new client
     end
 
-    # A Connections object, which lets disconnect clients from an OpenTok session.
+    # A Connections object, which lets you manage clients connected to an OpenTok session.
```
[the remark this answers is not in the document]
Comment (superchilled) [reply lib/opentok/opentok.rb:210]: Good catch! I think I changed that comment when I added the two new methods, but forgot to change it back when I subsequently removed them. Will update.

File: lib/opentok/streams.rb [hunk 3]
```diff
@@ -75,5 +75,35 @@ def layout(session_id, opts)
...
+    # @example
+    # {
+    #   "active": true,
+    #   "excluded_streams": [
+    #     "excludedStreamId1",
+    #     "excludedStreamId2"
+    #   ]
+    # }
+    #
+    def force_mute_all(session_id, opts = {})
```
Comment (jeffswartz) [inline lib/opentok/streams.rb:104]: I sort of think the `force_mute_all()` method should not be in the Streams class. It applies to all streams in a session, not to a specific stream. But i don't know where else to put it. We don't have an OpenTok::Sessions object. Wdyt, @superchilled? (I am not very proficient in Ruby, and if you think it's OK to have this method here, that's fine.)

**Review by jeffswartz — APPROVED** [review]

**Pull request merged**